File: Examples/test-suite/go/argcargvtest_runme.go
package main

import wrap "swigtests/argcargvtest"

func main() {
	largs := []string{"hi", "hola", "hello"}
	if ri := wrap.Mainc(largs); ri != 3 {
		panic(ri)
	}

	targs := []string{"hi", "hola"}
	if rs := wrap.Mainv(targs, 0); rs != "hi" {
		panic(rs)
	}
	if rs := wrap.Mainv(targs, 1); rs != "hola" {
		panic(rs)
	}
	if rs := wrap.Mainv(targs, 2); rs != "<<NULL>>" {
		panic(rs)
	}

// For dynamically typed languages we test this throws an exception or similar
// at runtime, but for Go this doesn't even compile (but we can't easily
// test for that here).
//	wrap.Mainv("hello", 1)

	wrap.InitializeApp(largs)

	// Check that an empty array works.
	empty_args := []string{};
	if ri := wrap.Mainc(empty_args); ri != 0 {
		panic(ri)
	}
	if rs := wrap.Mainv(empty_args, 0); rs != "<<NULL>>" {
		panic(rs)
	}

	// Check that empty strings are handled.
	empty_string := []string{"hello", "", "world"};
	if ri := wrap.Mainc(empty_string); ri != 3 {
		panic(ri)
	}
	if rs := wrap.Mainv(empty_string, 0); rs != "hello" {
		panic(rs)
	}
	if rs := wrap.Mainv(empty_string, 1); rs != "" {
		panic(rs)
	}
	if rs := wrap.Mainv(empty_string, 2); rs != "world" {
		panic(rs)
	}
	if rs := wrap.Mainv(empty_string, 3); rs != "<<NULL>>" {
		panic(rs)
	}
}

File: Examples/test-suite/csharp/argcargvtest_runme.cs
using System;
using argcargvtestNamespace;

public class argcargvtest_runme {

  public static void Main() {
    string[] largs = {"hi", "hola", "hello"};
    if (argcargvtest.mainc(largs) != 3)
      throw new Exception("bad main typemap");

    string[] targs = {"hi", "hola"};
    if (!argcargvtest.mainv(targs, 0).Equals("hi"))
      throw new Exception("bad main typemap");
    if (!argcargvtest.mainv(targs, 1).Equals("hola"))
      throw new Exception("bad main typemap");
    if (!argcargvtest.mainv(targs, 2).Equals("<<NULL>>"))
      throw new Exception("bad main typemap");

// For dynamically typed languages we test this throws an exception or similar
// at runtime, but for C# this doesn't even compile (but we can't easily
// test for that here).
//  argcargvtest.mainv("hello", 1);

    argcargvtest.initializeApp(largs);

    // Check that an empty array works.
    string[] empty_args = {};
    if (argcargvtest.mainc(empty_args) != 0)
      throw new Exception("bad main typemap");
    if (!argcargvtest.mainv(empty_args, 0).Equals("<<NULL>>"))
      throw new Exception("bad main typemap");

    // Check that empty strings are handled.
    string[] empty_string = {"hello", "", "world"};
    if (argcargvtest.mainc(empty_string) != 3)
      throw new Exception("bad main typemap");
    if (argcargvtest.mainv(empty_string, 0) != "hello")
      throw new Exception("bad main typemap");
    if (argcargvtest.mainv(empty_string, 1) != "")
      throw new Exception("bad main typemap");
    if (argcargvtest.mainv(empty_string, 2) != "world")
      throw new Exception("bad main typemap");
    if (argcargvtest.mainv(empty_string, 3) != "<<NULL>>")
      throw new Exception("bad main typemap");
  }
}
